llvm/lib/CodeGen/MachinePassManager.cpp

```diff
@@ -17,6 +17,51 @@
 
 using namespace llvm;
 
+namespace {
+
+/// RAII class to replace addMachinePrePasses/addMachinePostPasses
+/// If we need more complex logics, consider add class like
+/// CodeGenInstrumentation
+// TODO: skip FreeMachineFunctionPass and AsmPrinter
+class PrePostPasses {
```

Comment:
shouldn't this something that goes into `PIC`, rather than a separate RAII class? a random
```
PrePostPasses Hook(...);
```
in the code below goes against the entire purpose of pass instrumentation